### PR TITLE
Integrating master direct pacakge [BREAKING CHANGES]

### PR DESCRIPTION
**Pull request by kirankumarkolli**
Direct package internal types only (Only one single public type)
All REST contacts *Settigns Types are now open-sourced
o	Contracts code is simple and clean
o	No JsonSerializable or Resource/CosmosResource
o	You will see mix of older and newer contracts through-out (DocumentClient doesn’t use the OSS contracts)
o	Contract is limited to public surface area only
Json navigator code is open-sourced
o	No more references in master
o	All Json tests moved to unit-tests project
Some code moved back into direct package
o	Region and proximity util’s
o	Direct package only tests
o	StoreClientFactory

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj [hunk 1]
```diff
@@ -26,7 +26,7 @@
     <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
     <GenerateAssemblyInfo>true</GenerateAssemblyInfo>
     <GeneratePackageOnBuild>false</GeneratePackageOnBuild>
-    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <GenerateDocumentationFile>false</GenerateDocumentationFile>
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj]: Will be addressed in next PR

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Microsoft.Azure.Cosmos.EmulatorTests.csproj [hunk 2]
```diff
@@ -30,16 +30,7 @@
     <PackageReference Include="System.Reflection" Version="4.3.0" />
   </ItemGroup>
   
-  <ItemGroup Condition=" '$(SignAssembly)' == 'true' ">
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Microsoft.Azure.Cosmos.EmulatorTests.csproj]: Undo

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Microsoft.Azure.Cosmos.EmulatorTests.csproj]: Seems not required.

**Comment by kirankumarkolli** [comment]
All gates passed. One optional build-def is still running. But it never succeeded before. 
So for now merging it.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosAccountSettings.cs [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+    /// <summary> 
+    /// Represents a <see cref="CosmosAccountSettings"/>. A DatabaseAccountSettings is the container for databases in the Azure Cosmos DB service.
+    /// </summary>
+    public class CosmosAccountSettings
+    {
+        private Collection<CosmosAccountLocation> readLocations;
+        private Collection<CosmosAccountLocation> writeLocations;
+
+        internal readonly Lazy<IDictionary<string, object>> QueryEngineConfiurationInternal;
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosAccountSettings.cs]: Nit: QueryEngineConfigurationInternal

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosAccountSettings.cs [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+        internal long ReservedDocumentStorageInMB
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The provisioned documented storage capacity for the database account
+        /// Value is returned from cached information which is updated periodically and is not guaranteed to be real time
+        /// TODO remove this property tfs 4442779
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosAccountSettings.cs]: Does this tfs comment make any sense?

File: Native/Microsoft.Azure.Cosmos.Direct.targets [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (C) Microsoft Corporation. All rights reserved.
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    
+    <ItemGroup>
+      <ContentWithTargetPath Include="$(MSBuildThisFileDirectory)..\runtimes\win7-x64\native\Microsoft.Azure.Cosmos.ServiceInterop.dll">
```
Comment (ealsur) [inline Native/Microsoft.Azure.Cosmos.Direct.targets]: Is this inline with what @ausfeldt was doing?

**Pull request merged**